hoomd/device.py

```diff
@@ -316,30 +316,33 @@ def __init__(self, nthreads=None, communicator=None, msg_file=None, shared_msg_f
                                                            self.cpp_msg)
 
 
-class Auto(_device):
+def auto(communicator=None, msg_file=None, shared_msg_file=None, notice_level=2):
     """
     Allow simulation hardware to be chosen automatically by HOOMD-blue
 
     Args:
-        nthreads (int): number of TBB threads
         communicator (:py:mod:`hoomd.comm.Communicator`): MPI communicator object. Can be left None if using a
             default MPI communicator
         msg_file (str): Name of file to write messages to
         shared_msg_file (str): (MPI only) Name of shared file to write message to (append partition #)
         notice_level (int): Minimum level of notice messages to print
 
-    TODO: convert this to a function that produces a GPU or CPU device.
-    """
-
-    def __init__(self, nthreads=None, communicator=None, msg_file=None, shared_msg_file=None, notice_level=2):
-
-        _device.__init__(self, communicator, notice_level, msg_file, shared_msg_file)
 
-        _init_nthreads(nthreads)
+    Returns:
+        a GPU or CPU device, depending on availability, GPU is preferred
+    """
 
-        self.cpp_exec_conf = _hoomd.ExecutionConfiguration(_hoomd.ExecutionConfiguration.executionMode.AUTO,
+    self=_device(communicator, notice_level, msg_file, shared_msg_file)
```

Comment:
Python style reserves the `self` variable for class instance variables. This should be named `device` or something like that.

Reply:
I was nervous about making a variable the same name as the class and changed it. I will change it back.